packages/app/src/specs/SpecsList.vue

```diff
@@ -33,17 +79,44 @@
     />
     <div
       v-if="specs.length"
-      class="mb-4 grid grid-cols-2 children:font-medium children:text-gray-800 "
+      class="mb-4 grid grid-cols-7 children:font-medium children:text-gray-800"
+      :style="`padding-right: ${scrollbarOffset}px`"
     >
       <div
-        class="flex items-center justify-between"
+        class="flex col-span-3 items-center justify-between"
         data-cy="specs-testing-type-header"
       >
         {{ props.gql.currentProject?.currentTestingType === 'component' ?
           t('specPage.componentSpecsHeader') : t('specPage.e2eSpecsHeader') }}
       </div>
-      <div class="flex items-center justify-between">
-        <div>{{ t('specPage.lastUpdatedHeader') }}</div>
+      <div class="flex col-span-2 items-center justify-between">
+        <LastUpdatedHeader :is-git-available="isGitAvailable" />
+      </div>
+      <div class="flex items-center justify-end truncate">
+        <SpecHeaderCloudDataTooltip
+          :gql="props.gql"
+          :header-text-key-path="'specPage.latestRuns.header'"
+          :connected-text-key-path="'specPage.latestRuns.tooltip.connected'"
+          :not-connected-text-key-path="'specPage.latestRuns.tooltip.notConnected'"
+          :no-access-text-key-path="'specPage.latestRuns.tooltip.noAccess'"
+          :docs-text-key-path="'specPage.latestRuns.tooltip.linkText'"
+          :docs-url="'https://on.cypress.io/specs-latest-runs?utm_medium=Specs+Latest+Runs+Tooltip&utm_campaign=Latest+Runs'"
+          @showLogin="()=>showLogin('Specs Latest Runs Tooltip')"
+          @showConnectToProject="showConnectToProject"
+        />
+      </div>
+      <div class="flex items-center justify-end truncate">
+        <SpecHeaderCloudDataTooltip
+          :gql="props.gql"
+          :header-text-key-path="'specPage.averageDuration.header'"
+          :connected-text-key-path="'specPage.averageDuration.tooltip.connected'"
+          :not-connected-text-key-path="'specPage.averageDuration.tooltip.notConnected'"
+          :no-access-text-key-path="'specPage.averageDuration.tooltip.noAccess'"
+          :docs-text-key-path="'specPage.averageDuration.tooltip.linkText'"
+          :docs-url="'https://on.cypress.io/specs-average-duration?utm_medium=Specs+Average+Duration+Tooltip&utm_campaign=Average+Duration'"
+          @showLogin="()=>showLogin('Specs Average Duration Tooltip')"
+          @showConnectToProject="showConnectToProject"
+        />
```

Comment:
This template is quite large and complex at this point, which indicates to me we could make a few more components out of it. Not sure if changes like that are in scope at this point, but we should consider breaking this up before adding yet more features going forward.